sklearn/neighbors/tests/test_kde.py

```diff
@@ -141,3 +141,44 @@ def test_kde_pipeline_gridsearch():
     search = GridSearchCV(pipe1, param_grid=params, cv=5)
     search.fit(X)
     assert_equal(search.best_params_['kerneldensity__bandwidth'], .1)
+
+
+def test_kde_sample_weights():
+    N = 10000
+    T = 100
```

Comment:
Please avoid using single-letter variables (except `X`).
example: change `N` to `n_samples`.